src/app/core/models/configuration-event.ts

```diff
@@ -0,0 +1,12 @@
+import { TypedEvent } from '../../event-system/abstractions/event-observer.service'
+import { ConfigurationEventType } from './configuration-event-type'
+import { ShelfConfiguration } from '../../shared/models/shelf-configuration'
+
+export interface ConfigurationEvent extends TypedEvent {
+  type: ConfigurationEventType
+}
+
+export interface ShelfConfigurationUpdatedEvent extends ConfigurationEvent {
+  type: ConfigurationEventType.SHELF_UPDATED
+  shelf: ShelfConfiguration
+}
```

Comment:
The attributes here can be made readonly to enforce the functional needed for data mutations to be reactive.

Reply:
Updated